projects/plugins/jetpack/modules/google-fonts.php

```diff
@@ -115,6 +123,63 @@ function jetpack_add_google_fonts_provider() {
 }
 add_action( 'after_setup_theme', 'jetpack_add_google_fonts_provider' );
 
+/**
+ * Updates font names by filtering the data provided by the theme for global styles & settings.
+ *
+ * `wp_register_webfonts()` does not currently support giving a separate "name"
+ * from "font-family", even if theme.json schema itself does support it.
+ * See: https://github.com/WordPress/gutenberg/issues/46398
+ *
+ * Once support for "name" is added, below functionality can be refactored
+ * to provide the name directly within `wp_register_webfonts()`.
+ *
+ * @param WP_Theme_JSON_Data_Gutenberg $theme_json Class to access and update the underlying data.
+ * @return WP_Theme_JSON_Data_Gutenberg Class with updated Global Styles settings.
+ */
+function jetpack_rename_google_font_names( $theme_json ) {
+	$raw_data      = $theme_json->get_data();
+	$font_families = $raw_data['settings']['typography']['fontFamilies']['theme'];
+
+	$renamed_fonts = array(
+		'Alexandria'           => _x( 'Alexandria (Arabic)', 'Font name', 'jetpack' ),
+		'IBM Plex Sans Arabic' => _x( 'IBM Plex Sans (Arabic)', 'Font name', 'jetpack' ),
+		'Noto Sans Hebrew'     => _x( 'Noto Sans (Hebrew)', 'Font name', 'jetpack' ),
+		'Noto Sans HK'         => _x( 'Noto Sans (Hong Kong)', 'Font name', 'jetpack' ),
+		'Noto Sans JP'         => _x( 'Noto Sans (Japanese)', 'Font name', 'jetpack' ),
+		'Noto Sans KR'         => _x( 'Noto Sans (Korean)', 'Font name', 'jetpack' ),
+		'Noto Sans SC'         => _x( 'Noto Sans (Simplified Chinese)', 'Font name', 'jetpack' ),
+		'Noto Sans TC'         => _x( 'Noto Sans (Traditional Chinese)', 'Font name', 'jetpack' ),
+		'Noto Sans Telugu'     => _x( 'Noto Sans (Telugu)', 'Font name', 'jetpack' ),
+		'Noto Serif Hebrew'    => _x( 'Noto Serif (Hebrew)', 'Font name', 'jetpack' ),
+		'Noto Serif HK'        => _x( 'Noto Serif (Hong Kong)', 'Font name', 'jetpack' ),
+		'Noto Serif JP'        => _x( 'Noto Serif (Japanese)', 'Font name', 'jetpack' ),
+		'Noto Serif KR'        => _x( 'Noto Serif (Korean)', 'Font name', 'jetpack' ),
+		'Noto Serif SC'        => _x( 'Noto Serif (Simplified Chinese)', 'Font name', 'jetpack' ),
+		'Noto Serif TC'        => _x( 'Noto Serif (Traditional Chinese)', 'Font name', 'jetpack' ),
```

Comment:
@Automattic/i18n  Any difference to do above or use `sprintf`:

```php
$renamed_fonts = array(
    'Alexandria'           => sprintf( _x( '%s (Arabic)', 'Font name (font script)', 'jetpack' ), 'Alexandria'),
    'IBM Plex Sans Arabic' => sprintf( _x( '%s (Arabic)', 'Font name (font script)', 'jetpack' ), 'IBM Plex Sans'),
    'Noto Sans Hebrew'     => sprintf( _x( '%s (Hebrew)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Sans HK'         => sprintf( _x( '%s (Hong Kong)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Sans JP'         => sprintf( _x( '%s (Japanese)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Sans KR'         => sprintf( _x( '%s (Korean)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Sans SC'         => sprintf( _x( '%s (Simplified Chinese)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Sans TC'         => sprintf( _x( '%s (Traditional Chinese)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Sans Telugu'     => sprintf( _x( '%s (Telugu)', 'Font name (font script)', 'jetpack' ), 'Noto Sans'),
    'Noto Serif Hebrew'    => sprintf( _x( '%s (Hebrew)', 'Font name (font script)', 'jetpack' ), 'Noto Serif'),
    'Noto Serif HK'        => sprintf( _x( '%s (Hong Kong)', 'Font name (font script)', 'jetpack' ), 'Noto Serif'),
    'Noto Serif JP'        => sprintf( _x( '%s (Japanese)', 'Font name (font script)', 'jetpack' ), 'Noto Serif'),
    'Noto Serif KR'        => sprintf( _x( '%s (Korean)', 'Font name (font script)', 'jetpack' ), 'Noto Serif'),
    'Noto Serif SC'        => sprintf( _x( '%s (Simplified Chinese)', 'Font name (font script)', 'jetpack' ), 'Noto Serif'),
    'Noto Serif TC'        => sprintf( _x( '%s (Traditional Chinese)', 'Font name (font script)', 'jetpack' ), 'Noto Serif'),
);
```

Reply:
The current approach (without sprintf) should be better for localization and would allow translating the font name in some locales.

Reply:
I don't think the font name itself ever needs translating, only the part in brackets. Could be wrong tho!

Reply:
While some Wiki articles translate the 'Noto' family ([Arabic](https://ar.wikipedia.org/wiki/%D8%AE%D8%B7%D9%88%D8%B7_%D9%86%D9%88%D8%AA%D9%88), [Korean](https://ko.wikipedia.org/wiki/%EB%85%B8%ED%86%A0_%ED%8F%B0%ED%8A%B8)), the actual font name is displayed in English and most of translated versions of [Google Fonts FAQ](https://developers.google.com/fonts/faq) also use the English name for the font. The ones which do translate `Noto` (Hindi and Bengali) are not part of Mag-16 and have low translation rates on WordPress.com and Jetpack.

It should be safe to have the font names wrapped in translate calls and let the translators handle it, but since all of Mag-16 don't translate them and keeping it separate would allow to easily update the fonts, the `sprintf` approach looks better.

Reply:
Used `sprintf` instead in https://github.com/Automattic/jetpack/pull/27854/commits/42ff3e8a506a5d4a2bdad371cb74146c767337ef